From 147d28b975c9dee8e5c96a5698333fda4f33af68 Mon Sep 17 00:00:00 2001
From: Meir Levi <mlevi4@marvell.com>
Date: Thu, 24 Dec 2020 11:41:02 -0500
Subject: [PATCH 1070/1921] Documentation: devicetree: bindings:
 marvell-generic-uio documentation

Marvell generic UIO documentation

Signed-off-by: Meir Levi <mlevi4@marvell.com>
Change-Id: I35bedd9eecaca5fa27778ce4a7d0eac135126a4f
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/kernel/linux/+/42779
Tested-by: Stefan Chulski <stefanc@marvell.com>
Reviewed-by: Stefan Chulski <stefanc@marvell.com>
[WK: The original patch got from Marvell sdk11.21.09]
Signed-off-by: Wenlin Kang <wenlin.kang@windriver.com>
---
 .../devicetree/bindings/marvell-generic-uio.txt  | 16 ++++++++++++++++
 1 file changed, 16 insertions(+)
 create mode 100644 Documentation/devicetree/bindings/marvell-generic-uio.txt

diff --git a/Documentation/devicetree/bindings/marvell-generic-uio.txt b/Documentation/devicetree/bindings/marvell-generic-uio.txt
new file mode 100644
index 000000000000..f51b9c918f85
--- /dev/null
+++ b/Documentation/devicetree/bindings/marvell-generic-uio.txt
@@ -0,0 +1,16 @@
+* Marvell generic UIO
+
+Required properties:
+
+- compatible: should be "marvell,generic-uio".
+- reg: reg address
+- reg-name: reg name
+
+Example:
+
+CP11X_LABEL(uio_ethernet): EVALUATOR(uio_pp_, CP11X_NUM)@0 {
+		compatible = "generic-uio";
+		reg = <0x0 0x90000>, <0x130000 0x6000>,
+				<0x220000 0x1000>;
+		reg-names = "pp", "mspg", "cm3";
+};
-- 
2.31.1

